code/modules/antagonists/nukeop/equipment/nuclearbomb.dm

```diff
@@ -578,11 +578,18 @@
 /obj/machinery/nuclearbomb/beer/really_actually_explode()
 	disarm()
 
+
+/**
+  * Kills any mob on the z-level, assuming they're not in a freezer
+  *
+  * Arguments:
+  * * z - the z-level to kill people on
+  */
 /proc/KillEveryoneOnZLevel(z)
 	if(!z)
 		return
 	for(var/mob/M in GLOB.mob_list)
-		if(M.stat != DEAD && M.z == z)
+		if(M.stat != DEAD && M.z == z && !istype(M.loc, /obj/structure/closet/secure_closet/freezer))
```

Comment:
You realize that if you are in a freezer, you are in nullspace, so you dont exist on the Z level making your check redundant as the M.z == z check will prevent anyone in a locker or other container from dying, perhaps you might wanna fix that?
```suggestion
		if(M.stat != DEAD && get_turf(M).z == z && !istype(M.loc, /obj/structure/closet/secure_closet/freezer))
```